<launch>
    <!-- run the px4_vision_pose.cpp -->
    <node pkg="px4ctrl" type="px4_pos_estimator" name="px4_pos_estimator" output="screen">
        <!-- 定位数据输入源: 6 - VICON, 2 - Gazebo, 3 - T265, 5 - LIO, 0 - optitrack, 1 - VIO -->
        <param name="input_source" value="2"/>
        <param name="LIO_topic" value="/Odometry"/>
        <param name="Gazebo_topic" value="/gazebo/groundtruth/odom"/>
        <param name="T265_topic" value="/t265/odom/sample"/>
        <param name="VIO_topic" value="/vins_estimator/odometry"/>
    </node>

    <!-- 启动uav_control_main -->
    <node pkg="px4ctrl" type="px4ctrl" name="px4ctrl"
          output="screen" launch-prefix="gnome-terminal --">
        <rosparam command="load" file="$(find px4ctrl)/config/uav_control_indoor.yaml"/>
    </node>

        <!-- 启动虚拟摇杆驱动 -->
    <node pkg="px4ctrl" type="joy_node" name="joy_node" output="screen"/>

    <!-- run the terminal_control.cpp -->
    <node pkg="px4ctrl" type="terminal_control" name="terminal_control"
          output="screen" launch-prefix="gnome-terminal --">
        <rosparam command="load" file="$(find px4ctrl)/config/uav_control_indoor.yaml"/>
    </node>

    <node pkg="px4ctrl" type="direct_goal" name="direct_goal"/>
</launch>
